MastodonSDK/Sources/MastodonCore/AuthenticationServiceProvider.swift

```diff
@@ -101,10 +101,29 @@ public extension AuthenticationServiceProvider {
             logger.log(level: .error, "Could not migrate legacy authentications")
         }
     }
-    
+
     var authenticationMigrationRequired: Bool {
         userDefaults.didMigrateAuthentications == false
     }
+
+    func fetchAccounts(apiService: APIService, completion: (() -> Void)? = nil) async {
+        // FIXME: This is a dirty hack to make the performance-stuff work.
+        // Problem is, that we don't persist the user on disk anymore. So we have to fetch
+        // it when we need it to display on the home timeline.
+        // We need this (also) for the Account-list, but it might be the wrong place. App Startup might be more appropriate
+        for authentication in authentications {
+            guard let account = try? await apiService.accountInfo(domain: authentication.domain,
+                                                                  userID: authentication.userID,
+                                                                  authorization: Mastodon.API.OAuth.Authorization(accessToken: authentication.userAccessToken)).value else { continue }
+
+            FileManager.default.store(account: account, forUserID: authentication.userIdentifier())
+        }
+
+        NotificationCenter.default.post(name: .userFetched, object: nil)
+
+        completion?()
```

Comment:
Maybe we can get rid of this closure and use the `.userFetched`notification instead?

Reply:
We can get rid of `completion` as we don't use that anymore. Good catch 👍